File: java/com/lc/lc24.java
package lc;

public class lc24 {
    public ListNode swapPairs(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        ListNode dummy = new ListNode(-1, head);
        ListNode pre = dummy, slow = dummy.next, fast = dummy.next.next, temp;
        while (fast != null) {
            temp = fast.next;
            pre.next = fast;
            fast.next = slow;
            slow.next = temp;
            if (temp == null) {
                break;
            }
            pre = slow;
            slow = temp;
            fast = slow.next;
        }
        return dummy.next;
    }
}

File: go/lc/lc24.go
package main

func swapPairs(head *ListNode) *ListNode {
	dummyHead := &ListNode{Next: head}
	cur := dummyHead
	for cur.Next != nil && cur.Next.Next != nil {
		fi := cur.Next
		se := cur.Next.Next

		cur.Next = se
		fi.Next = se.Next
		se.Next = fi

		cur = fi
	}
	return dummyHead.Next
}
